src/state/slices/opportunitiesSlice/selectors.ts

```diff
@@ -403,25 +387,25 @@ export const selectAggregatedEarnUserStakingOpportunitiesIncludeEmpty =
 // TODO: testme
 export const selectAggregatedEarnUserStakingOpportunity = createDeepEqualOutputSelector(
   selectAggregatedEarnUserStakingOpportunities,
-  (earnOpportunities): StakingEarnOpportunityType | undefined =>
+  selectAssets,
+  (earnOpportunities, assets): StakingEarnOpportunityType | undefined =>
     earnOpportunities.reduce<StakingEarnOpportunityType | undefined>((acc, currentOpportunity) => {
+      const asset = assets[currentOpportunity.assetId]
+      const underlyingAsset = assets[currentOpportunity.underlyingAssetId]
+
       return Object.assign({}, acc, currentOpportunity, {
         cryptoAmountBaseUnit: bnOrZero(currentOpportunity.stakedAmountCryptoBaseUnit)
+          .div(bn(10).pow(asset?.precision ?? underlyingAsset?.precision))
           .plus(acc?.stakedAmountCryptoBaseUnit ?? 0)
           .toString(),
-        fiatAmount: bnOrZero(currentOpportunity?.rewardsAmountsCryptoPrecision?.[0])
-          .plus(acc?.rewardsAmountsCryptoPrecision?.[0] ?? 0)
+        fiatAmount: bnOrZero(currentOpportunity?.rewardsAmountsCryptoBaseUnit?.[0])
+          .div(bn(10).pow(asset?.precision ?? underlyingAsset?.precision))
+          .plus(acc?.rewardsAmountsCryptoBaseUnit?.[0] ?? 0)
+          .div(bn(10).pow(asset?.precision ?? underlyingAsset?.precision))
```

Comment:
double division? my BODMAS is hurting. add base units and divide once?

Reply:
7d59baac6786376fa14f13bd8117f9b8e457f78